src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetAnnotationExecutor.java

```diff
@@ -79,7 +91,27 @@ private boolean runLoaddatasetid(ExternalDataBaseEntityService externalDataBaseE
 				addException(history, new ObjectUpdateExceptionData(dto, e.getMessage(), e.getStackTrace()));
 			}
 		}
-		ph.finishProcess();
-		return true;
+		updateHistory(history);
+		ph.progressProcess();
+	}
+
+	private void runLoadHtpAnnotations(HTPExpressionDatasetAnnotationService htpExpressionDatasetAnnotationService, BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<HTPExpressionDatasetAnnotationFmsDTO> htpDatasetData, List<Long> annotationsIdsLoaded, ProcessDisplayHelper ph) {
+		for (HTPExpressionDatasetAnnotationFmsDTO dto : htpDatasetData) {
+			try {
+				HTPExpressionDatasetAnnotation dbObject = htpExpressionDatasetAnnotationService.upsert(dto, dataProvider);
+				history.incrementCompleted();
+				if (annotationsIdsLoaded != null) {
+					annotationsIdsLoaded.add(dbObject.getId());
+				}
+			} catch (ObjectUpdateException e) {
+				history.incrementFailed();
+				addException(history, e.getData());
+			} catch (Exception e) {
+				history.incrementFailed();
```

Comment:
Probably want to print stackTrace here too otherwise unexpected exceptions that are caught don't get printed in the logs - same goes for the above runLoaddatasetid method.

Reply:
Sure, will print them in both the methods